src/wasm-type.h

```diff
@@ -263,11 +263,14 @@ class Type {
 
   std::string toString() const;
 
-  struct Iterator : std::iterator<std::random_access_iterator_tag,
-                                  Type,
-                                  long,
-                                  Type*,
-                                  const Type&> {
+  struct Iterator {
+    // Iterator traits
+    using iterator_category = std::random_access_iterator_tag;
+    using value_type = Type;
+    using difference_type = long;
```

Comment:
actually, should this just be `ptrdiff_t` instead of `long`?

Reply:
yep, probably.